Source/Core/VideoBackends/D3D/TextureCache.cpp

```diff
@@ -196,34 +196,34 @@ TextureCache::TextureCache()
   auto lutBd = CD3D11_BUFFER_DESC(sizeof(u16) * 256, D3D11_BIND_SHADER_RESOURCE);
   HRESULT hr = D3D::device->CreateBuffer(&lutBd, nullptr, &palette_buf);
   CHECK(SUCCEEDED(hr), "create palette decoder lut buffer");
-  D3D::SetDebugObjectName(palette_buf, "texture decoder lut buffer");
+  D3D::SetDebugObjectName(palette_buf.Get(), "texture decoder lut buffer");
   // TODO: C14X2 format.
   auto outlutUavDesc =
-      CD3D11_SHADER_RESOURCE_VIEW_DESC(palette_buf, DXGI_FORMAT_R16_UINT, 0, 256, 0);
-  hr = D3D::device->CreateShaderResourceView(palette_buf, &outlutUavDesc, &palette_buf_srv);
+      CD3D11_SHADER_RESOURCE_VIEW_DESC(palette_buf.Get(), DXGI_FORMAT_R16_UINT, 0, 256, 0);
+  hr = D3D::device->CreateShaderResourceView(palette_buf.Get(), &outlutUavDesc, &palette_buf_srv);
   CHECK(SUCCEEDED(hr), "create palette decoder lut srv");
-  D3D::SetDebugObjectName(palette_buf_srv, "texture decoder lut srv");
+  D3D::SetDebugObjectName(palette_buf_srv.Get(), "texture decoder lut srv");
   const D3D11_BUFFER_DESC cbdesc =
       CD3D11_BUFFER_DESC(16, D3D11_BIND_CONSTANT_BUFFER, D3D11_USAGE_DEFAULT);
   hr = D3D::device->CreateBuffer(&cbdesc, nullptr, &palette_uniform);
   CHECK(SUCCEEDED(hr), "Create palette decoder constant buffer");
-  D3D::SetDebugObjectName((ID3D11DeviceChild*)palette_uniform,
+  D3D::SetDebugObjectName(palette_uniform.Get(),
                           "a constant buffer used in TextureCache::CopyRenderTargetToTexture");
 }
 
 TextureCache::~TextureCache()
 {
   for (unsigned int k = 0; k < MAX_COPY_BUFFERS; ++k)
-    SAFE_RELEASE(s_efbcopycbuf[k]);
+    s_efbcopycbuf[k].Reset();
 
   g_encoder->Shutdown();
   g_encoder.reset();
 
-  SAFE_RELEASE(palette_buf);
-  SAFE_RELEASE(palette_buf_srv);
-  SAFE_RELEASE(palette_uniform);
-  for (ID3D11PixelShader*& shader : palette_pixel_shader)
-    SAFE_RELEASE(shader);
+  palette_buf.Reset();
+  palette_buf_srv.Reset();
+  palette_uniform.Reset();
+  for (ComPtr<ID3D11PixelShader>& shader : palette_pixel_shader)
+    shader.Reset();
```

Comment:
Is the order significant here? If not, these `Reset()` calls could probably be removed?